.github/workflows/MacOs.yaml

```diff
@@ -110,17 +109,19 @@ jobs:
           source ./spack/share/spack/setup-env.sh
           spack env activate spectre
           spack buildcache create -uf -m dependencies $SPECTRE_SPACK_DEPS
-      # Install Charm++ separately. We may be able to install it via Spack as
-      # well and patch it.
+      # Install Charm++ separately. We can install it via Spack once a package
+      # version greater than 6.10.2 is available, since that fixes
+      # https://github.com/sxs-collaboration/spectre/issues/2397.
+      # (installing charmpp@develop via Spack is currently broken, see
+      # https://github.com/spack/spack/issues/18460)
       - name: Install Charm++
         if: steps.restore-dependencies.outputs.cache-hit != 'true'
         run: |
           cd $HOME/dependencies
-          wget http://charm.cs.illinois.edu/distrib/charm-${CHARM_VERSION}.tar.gz
-          tar -xzf charm-${CHARM_VERSION}.tar.gz
-          cd ./charm-v${CHARM_VERSION}
-          ./build charm++ multicore-darwin-x86_64 -j4 -O0 $SPECTRE_MIN_MACOS
-          rm ../charm-${CHARM_VERSION}.tar.gz
+          git clone https://github.com/UIUC-PPL/charm && cd charm
+          # This is the commit that fixed #2397
+          git checkout e7e9d4b5b27608b89ab09f021b6b5194267f74fe
```

Comment:
I'm confused, why does this install an old version of Boost to avoid the issue, but then check out a random hash to avoid the boost issue... I feel like we should only check out the exact Charm++ version we support

Reply:
#2397 has nothing to do with boost. Some `.tbd` files appeared for some reason on GitHub's latest macOS VM and Charm couldn't handle those files, so our CI build broke. This commit https://github.com/UIUC-PPL/charm/commit/e7e9d4b5b27608b89ab09f021b6b5194267f74fe is the one that fixes the issue. It will be in charm v6.11, but they'll probably not have another bugfix release before that. So I figured it's better to checkout the particular commit than to checkout whatever the latest `master` is when the CI build runs. It's the closest thing to a bugfix release we can do I think.